// Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
//
// NVIDIA CORPORATION and its licensors retain all intellectual property
// and proprietary rights in and to this software, related documentation
// and any modifications thereto.  Any use, reproduction, disclosure or
// distribution of this software and related documentation without an express
// license agreement from NVIDIA CORPORATION is strictly prohibited.


syntax = "proto3";

package nvidia.riva.tts;

option cc_enable_arenas = true;
option go_package = "nvidia.com/riva_speech";

import "riva/proto/riva_audio.proto";

service RivaSpeechSynthesis {
    // Used to request text-to-speech from the service. Submit a request containing the
    // desired text and configuration, and receive audio bytes in the requested format.
    rpc Synthesize(SynthesizeSpeechRequest) returns (SynthesizeSpeechResponse) {}

    // Used to request text-to-speech returned via stream as it becomes available.
    // Submit a SynthesizeSpeechRequest with desired text and configuration,
    // and receive stream of bytes in the requested format.
    rpc SynthesizeOnline(SynthesizeSpeechRequest) returns (stream SynthesizeSpeechResponse) {}
}

message SynthesizeSpeechRequest {
    string text = 1;
    string language_code = 2;
    // audio encoding params
    AudioEncoding encoding = 3;
    int32 sample_rate_hz = 4;
    // voice params
    string voice_name = 5;
}

message SynthesizeSpeechResponseMetadata {
    // Currently experimental API addition that returns the input text
    // after preprocessing has been completed as well as the predicted
    // duration for each token.
    // Note: this message is subject to future breaking changes, and potential
    // removal.
    string text = 1;
    string processed_text = 2;
    repeated float predicted_durations = 8;
}

message SynthesizeSpeechResponse {
    bytes audio = 1;
    SynthesizeSpeechResponseMetadata meta = 2;
}

/*
 *
 */
